pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentDispatcherSingleActiveConsumer.java

```diff
@@ -458,29 +454,26 @@ protected Pair<Integer, Long> calculateToRead(Consumer consumer) {
         return Pair.of(messagesToRead, bytesToRead);
     }
 
-    @Override
-    public void readEntriesFailed(ManagedLedgerException exception, Object ctx) {
-        executor.execute(() -> internalReadEntriesFailed(exception, ctx));
-    }
-
-    private synchronized void internalReadEntriesFailed(ManagedLedgerException exception, Object ctx) {
+    @VisibleForTesting
+    public synchronized Void readEntriesFailed(Throwable throwable, Consumer consumer) {
```

Comment:
what is the reason for using `Void` return type?

Reply:
It doesn't seem to be a good idea to use `Void` return type for public methods.

Reply:
I've updated the code, PTAL again